lib/Controller/EndpointController.php

```diff
@@ -229,7 +229,7 @@ public function deleteNotification(int $id): DataResponse {
 			$deleted = $this->handler->deleteById($id, $this->getCurrentUser(), $notification);
 
 			if ($deleted) {
-				$this->push->pushDeleteToDevice($this->getCurrentUser(), [$id], $notification->getApp());
+				$this->manager->notifyDelete($this->getCurrentUser(), $id, $notification);
```

Comment:
I think you'd like to be always notified, not only when using the explicit delete?
But also e.g. when someone enters a Talk chat which marks all mention notifications read when they scroll down, Or when they join a call the call notification is removed.

This basically means you would want to be an NotificationApp yourself and listen to the `markProcessed` handling.

It's a bit weirdly abstracted and not clear (also not aware), but technically it is not required that the notifications app is not the only notification app. That's exactly why the API never operates based on IDs.
I guess you'd like to "be part of it" and basically want to act based on this app's notification id?
Similarly how we do it for the existing Push class: https://github.com/nextcloud/notifications/blob/0b63d0e073889223e1419cf01b6df016276ca23a/lib/App.php#L70-L74

Best idea that comes to my mind would be to emit an event in this app here with all deleted IDs?